### PR TITLE
Refactor: Improve location compare

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by thorn0 — APPROVED** [review]

**Review by evilebottnawi — APPROVED** [review]

File: src/main/comments.js [hunk 1]
```diff
@@ -103,19 +105,18 @@ function decorateComment(node, comment, options) {
...
+    if (start <= commentStart && commentEnd <= end) {
       // The comment is completely contained by this child node.
       comment.enclosingNode = child;
 
       decorateComment(child, comment, options);
       return; // Abandon the binary search at this level.
     }
 
-    if (locEnd(child) - locStart(comment) <= 0) {
+    if (start <= commentStart) {
```
Comment (thorn0) [inline src/main/comments.js:119]: Ouch! Should be `end <= commentStart`.

Comment (fisker) [reply src/main/comments.js:119]: Ah...sorry about this.

Comment (thorn0) [reply src/main/comments.js:119]: I pushed a fix 1344d9a18a

Comment (fisker) [reply src/main/comments.js:119]: Thank you!

**Pull request merged**